src/python/pants/pantsd/pailgun_server.py

```diff
@@ -270,13 +273,16 @@ def yield_and_release(time_waited):
         yield
     else:
       # We have to wait for another request to finish being handled.
-      NailgunProtocol.send_stderr(request, "Another pants invocation is running. Will wait {} for it to finish before giving up.\n".format(
+      self._send_stderr(request, "Another pants invocation is running. Will wait {} for it to finish before giving up.\n".format(
         "forever" if self._should_poll_forever(timeout) else "up to {} seconds".format(timeout)
       ))
+      self._send_stderr(request, "If you want to have multiple pants invocations at once, please "
```

Comment:
Let's have this refer to setting the env var in "one of the environments": because it might not be "this" one that should not use pantsd: rather, the other one.

Also, rather than "if you want to have", maybe "if you don't want to wait for the first run to finish" (ie, implying that the workaround is not something that applies in all situations: just something you do for the second run).

Reply:
I've implemented the second suggestion, but I think that ends up implying the first, if we just say that
> another pants invocation is running with pantsd enabled

as we do now. I'm not sure if there's an intricacy that's missing?

Reply:
The intricacy is that maybe the _other_ run is the one that should be run without pantsd. So it might be something like "kill one of the runs and re-run it with `$ENV`". Up to you.